platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -831,6 +832,17 @@ public class SwirldsPlatform implements Platform {
         GuiPlatformAccessor.getInstance().setLatestImmutableStateComponent(selfId, latestImmutableState);
     }
 
+    /**
+     * Wait until the consensus round handler is not busy.
+     */
+    private void waitUntilTransactionHandlingThreadIsNotBusy() {
+        try {
+            consensusRoundHandler.waitUntilNotBusy();
+        } catch (final InterruptedException e) {
+            throw new RuntimeException("Interrupted waiting for transaction handling thread to not be busy", e);
+        }
```

Comment:
```suggestion
        } catch (final InterruptedException e) {
            Thread.currentThread().interrupt();
            throw new RuntimeException("Interrupted waiting for transaction handling thread to not be busy", e);
        }
```

It's best practice to always re-interrupt the current thread if you catch an interrupted exception and rethrow it as something else.

Reply:
Done